### DISCUSSION TITLE
How to pass input value to other script?

### DISCUSSION THREAD

**Discussion by Tim-Cao** [Textualize/textual#3144] [Q&A] [answered]
I would like to create a Terminal UI using Textual for my scripts, for example
https://github.com/Tim-Cao/work-with-script/blob/master/script/create_test_fix_ticket.py

How to pass value from yield Input() then use Button invoke my scripts?

```
from textual.app import App, ComposeResult
from textual.binding import Binding
from textual.widgets import Button, Footer, Input, Label, TabbedContent, TabPane

class ScriptApp(App):

    BINDINGS = [
        Binding(key="ctrl+q", action="quit", description="Quit the app"),
        Binding(key="pagedown", action="down", description="Scroll down")
    ]

    def compose(self) -> ComposeResult:
        with TabbedContent():
            with TabPane("Forward Failure PR", id="tab-1"):
                yield Label("Failure Pull Request Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Create PR And Forward", id="tab-2"):
                yield Label("Local Branch Name: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Create TF Ticket", id="tab-3"):
                yield Label("Testray Case Result Id: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Write Comments", id="tab-4"):
                yield Label("Jira Ticket Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Write Description", id="tab-5"):
                yield Label("Jira Ticket Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
        yield Footer()

app = ScriptApp()

if __name__ == "__main__":
    app.run()
```

**Comment by davep** [comment, marked as answer]
By the sounds of things you want to be looking either (or both, depending on how you want to do things) the [`Input.Submitted`](https://textual.textualize.io/widgets/input/#textual.widgets._input.Input.Submitted) message and the [`Button.Pressed`](https://textual.textualize.io/widgets/button/#textual.widgets._button.Button.Pressed) message. If you are unfamiliar with how events and messages work [there's a good guide to that](https://textual.textualize.io/guide/events/), and of course they are [introduced in the tutorial](https://textual.textualize.io/tutorial/).

**Comment by Tim-Cao** [comment]
I didn't find example about event calling another python script.
@davep, could you help me?

**Reply by davep** [reply]
It's not really clear to me what you're asking, sorry.

**Reply by Tim-Cao** [reply]
Let me explain,
I would like to create an app could interactive with other script.
```
from textual.app import App, ComposeResult
from textual.binding import Binding
from textual.widgets import Button, Footer, Input, Label, TabbedContent, TabPane

class ScriptApp(App):

    BINDINGS = [
        Binding(key="ctrl+q", action="quit", description="Quit the app"),
        Binding(key="pagedown", action="down", description="Scroll down")
    ]

    def compose(self) -> ComposeResult:
        with TabbedContent():
            with TabPane("Forward Failure PR", id="tab-1"):
                yield Label("Failure Pull Request Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Create PR And Forward", id="tab-2"):
                yield Label("Local Branch Name: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Create TF Ticket", id="tab-3"):
                yield Label("Testray Case Result Id: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Write Comments", id="tab-4"):
                yield Label("Jira Ticket Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
            with TabPane("Write Description", id="tab-5"):
                yield Label("Jira Ticket Number: ")
                yield Input()
                yield Label("Submit")
                yield Button()
        yield Footer()

app = ScriptApp()

if __name__ == "__main__":
    app.run()
 ```
 This is an app I created using Textual framework as a frontend
 
I don't know how to make it communicate with https://github.com/Tim-Cao/work-with-script/blob/master/script/create_test_fix_ticket.py
Pass the input to that script and return the output

**Reply by willmcgugan** [reply]
I think you want [subprocesses](https://docs.python.org/3/library/asyncio-subprocess.html). There should be plenty of examples on the web to follow.

**Reply by Tim-Cao** [reply]
No. I've made a demo works
https://github.com/Tim-Cao/work-with-script/pull/5/commits/36ce06883b1e25c872c25372257a4cfb7233b3b7#diff-4b16fce2f8e693ff4eedff786dc96babe8b4f1f928af59ccebb8f6a7cdb31170
![image](https://github.com/Textualize/textual/assets/52661397/4ecff04e-ef3b-4173-b299-ed0c8431524d)

**Reply by Tim-Cao** [reply]
@davep and @willmcgugan
How to throw the error shown on app not in Traceback?
![image](https://github.com/Textualize/textual/assets/52661397/88969ab5-2cd2-4caf-9fb8-0d09176cade0)

**Reply by willmcgugan** [reply]
Somewhere in your code you have done `int("abc")`. You will need to handle the ValueError in your code to prevent the app from exiting.

**Reply by Tim-Cao** [reply]
I just used native int() function, what should I do to prevent the app from exiting for any error?
And thrown them into app?

**Reply by willmcgugan** [reply]
Textual doesn't behave any differently from any other Python app. If your code throws an exception and you don't catch it, then the app will exit.

You might want to ask such questions in another forum for Python development. But if you have any Textual related questions, we're here to help.

**Reply by Tim-Cao** [reply]
I see. Thanks
